soh/src/overlays/actors/ovl_En_GirlA/z_en_girla.c

```diff
@@ -563,7 +596,7 @@ s32 EnGirlA_CanBuy_DekuShield(GlobalContext* globalCtx, EnGirlA* this) {
 }
 
 s32 EnGirlA_CanBuy_GoronTunic(GlobalContext* globalCtx, EnGirlA* this) {
-    if (LINK_AGE_IN_YEARS == YEARS_CHILD) {
+    if (LINK_AGE_IN_YEARS == YEARS_CHILD && !Randomizer_GetSettingValue(RSK_SHOPSANITY)) {
```

Comment:
i don't remember, does GetSettingValue have a 64ddcheck in it?

Reply:
Good catch, in fact, I think this will actually crash on switch if you're not playing a rando save, since the randoSettings list will be empty. Similar bug to what was crashing Switches when approaching bottleable things.

Reply:
It does not, good catch

Reply:
i think short term (for this PR) we should just add the 64dd checks next to these calls to GetSettingValue. we make enough calls to GetSettingValue throughout the codebase that we should verify that adding a 64dd check in there won't beak things, which could easily blow scope

Reply:
If there's not already one, I'll open an issue about cleaning up/clarifying the conditionals on the v3 milestone. Bare minimum I'd like to put gSaveContext.n64ddflag behind a macro so we can change it in the future if need be.

Reply:
https://github.com/HarbourMasters/Shipwright/issues/1490